crocket/src/main/java/com/crocket/CollisionHandler.java

```diff
@@ -1,45 +1,204 @@
-
-        //Normalise the normal vector 
-        double magnitude = Math.sqrt(dx * dx + dy * dy);
-        double unitNormalX = dx / magnitude;
-        double unitNormalY = dy / magnitude;
-        double unitTangentX = -unitNormalY;
-        double unitTangentY = unitNormalX;
-
-        //Project the normal
-        double v1n = unitNormalX * entity1.getxVelocity() + unitNormalY * entity1.getyVelocity();
-        double v2n = unitNormalX * entity2.getxVelocity() + unitNormalY * entity2.getyVelocity();
-
-        double v1t = unitTangentX * entity1.getxVelocity() + unitTangentY * entity1.getyVelocity();
-        double v2t = unitTangentX * entity2.getxVelocity() + unitTangentY * entity2.getyVelocity();
-
-        double mass1 = entity1.getMass();
-        double mass2 = entity2.getMass();
-
-        //Normal velocities with the one-dimensional collision formula
-        double newV1n = (v1n * (mass1 - mass2) + 2 * mass2 * v2n)/(mass1 + mass2);
-        double newV2n = (v2n * (mass2 - mass1) + 2 * mass1 * v1n)/(mass1 + mass2);
-
-        //Converts the scalar normal and tangential velocities into vectors 
-        //And makes final velocity vectors by adding the normal and tangential components.
-        double newXVelocityEntity1 = (newV1n * unitNormalX) + (v1t * unitTangentX);
-        double newXVelocityEntity2 = (newV2n * unitNormalX) + (v2t * unitTangentX);
-        double newYVelocityEntity1 = (newV1n * unitNormalY) + (v1t * unitTangentY);
-        double newYVelocityEntity2 = (newV2n * unitNormalY) + (v2t * unitTangentY);
-
-        entity1.setxVelocity(newXVelocityEntity1);
-        entity1.setyVelocity(newYVelocityEntity1);
-
-        entity2.setxVelocity(newXVelocityEntity2);
-        entity2.setyVelocity(newYVelocityEntity2);  
+    public static boolean intersect(Entity entity1, Entity entity2)  {
+        Hitbox e1Hitbox = entity1.getHitbox();
+        Hitbox e2Hitbox = entity2.getHitbox();
+
+        double e1X = e1Hitbox.getxPosition();
+        double e1Y = e1Hitbox.getyPosition();
+        int e1Width = e1Hitbox.getWidth();
+        int e1Height = e1Hitbox.getHeight();
+
+        double e2X = e2Hitbox.getxPosition();
+        double e2Y = e2Hitbox.getyPosition();
+        int e2Width = e2Hitbox.getWidth();
+        int e2Height = e2Hitbox.getHeight();
+
+        return (e1X < e2X + e2Width &&
+                e1X + e1Width > e2X &&
+                e1Y < e2Y + e2Height &&
+                e1Y + e1Height > e2Y);
+    }
+
+    public static boolean intersect(Entity entity, Hitbox hitbox) {
+        Hitbox eHitbox = entity.getHitbox();
+
+        double eX = eHitbox.getxPosition();
+        double eY = eHitbox.getyPosition();
+        int eWidth = eHitbox.getWidth();
+        int eHeight = eHitbox.getHeight();
+
+        double hX = hitbox.getxPosition();
+        double hY = hitbox.getyPosition();
+        int hWidth = hitbox.getWidth();
+        int hHeight = hitbox.getHeight();
+
+        return (eX < hX + hWidth &&
```

Comment:
intersect(Entity entity, Hitbox hitbox) and intersect(Hitbox hitbox1, Hitbox hitbox2) are very similar. Is there any way to reduce code duplication?

Reply:
Refactored this